### PR TITLE
Km/test suites test refactor

### PR DESCRIPTION
**Pull request by DPUkyle**
WIP on test coverage for forthcoming jvm-test-suite plugin.

### CODE REVIEW

File: subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    `jvm-test-suite`
+}
+
+repositories {
+    mavenCentral()
+}
+
+testing {
+    suites {
+        named<JvmTestSuite>("unitTest") { // FIXME TestSuite with name 'unitTest' not found.
```
Comment (big-guy) [inline subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts:12]: The name is "test" and not "unitTest" now

File: subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+plugins {
+    java
+    `jvm-test-suite`
```
Comment (big-guy) [inline subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts:3]: this is applied by java

File: subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    mavenCentral()
+}
+
+testing {
+    suites {
+        named<JvmTestSuite>("unitTest") { // FIXME TestSuite with name 'unitTest' not found.
+            useJUnit()
+            dependencies {
+                implementation("junit:junit:4.13")
+            }
```
Comment (big-guy) [inline subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts:16]: with `useJUnit()` we add this dependency already

File: subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            targets {
+                all {
+                    // tag::test-categories[]
+                    testTask.configure {
+                        options { // FIXME Ugly unsafe casts. Do we need to qualify the type of options below?
```
Comment (big-guy) [inline subprojects/docs/src/snippets/testing/junit-test-suite-categories/kotlin/build.gradle.kts:21]: ugh, I didn't think about how Kotlin would hate this.

We can just use `useJUnit {` here instead. That should get rid of the cast

File: subprojects/docs/src/snippets/testing/junit-test-suite-categories/groovy/build.gradle [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    id 'jvm-test-suite'
+}
+
+repositories {
+    mavenCentral()
+}
+
+testing {
+    suites {
+        unitTest {
```
Comment (big-guy) [inline subprojects/docs/src/snippets/testing/junit-test-suite-categories/groovy/build.gradle:12]: the built-in test is under "test" and not "unitTest" now

File: subprojects/docs/src/snippets/testing/junit-test-suite-categories/groovy/build.gradle [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+repositories {
+    mavenCentral()
+}
+
+testing {
+    suites {
+        test {
+            useJUnit()
+
+            dependencies {
```
Comment (tresat) [inline subprojects/docs/src/snippets/testing/junit-test-suite-categories/groovy/build.gradle:14]: This should be unnecessary, the useJUnit() method should add the deps automatically.

**Review by tresat — APPROVED** [review]

**Pull request closed without merging**